076-support-for-kafka-4.0.md

```diff
@@ -0,0 +1,203 @@
+   This version might add support for some of the new features added to KRaft in Kafka 3.9.0.
+   0.4y will keep the support for 3.8 and 3.9 as well as support for both KRaft and ZooKeeper.
+   This version will be subject to the extended support described below in a dedicated section.
+   It will also keep the warnings introduced in Strimzi 0.4x.
+
+3) Once Strimzi 0.4y is released, we will proceed and remove ZooKeeper support from Strimzi.
+   This would include things such as:
+   * Deprecating the ZooKeeper related parts of the CRDs
+   * Removing the ZooKeeper reconciler and all other parts of the code relevant to ZooKeeper
+   * Remove the migration code
+   * Remove ZooKeeper related system tests
+   * Remove ZooKeeper related content from docs
+
+   As upgrade to Strimzi 0.4z will not be possible without migration to KRaft, we will know for sure that the clusters run by Strimzi 0.4z had to be previously run by recent Strimzi versions.
+   That will also allow us to remove other legacy code.
+   For example the migration of Kafka clusters from StatefulSets to StrimziPodSets would not be needed anymore and would be removed.
+
+   Once Apache Kafka 4.0 is released, we will adopt it and do a Strimzi release 0.4z.
+   0.4z and all later Strimzi versions will support only KRaft based clusters.
+   Strimzi 0.4z will support the last two Kafka minor versions, as usual - Kafka 3.9 and 4.0 - and will allow users to gradually upgrade as always.
+   But it will support only KRaft and users with ZooKeeper-based clusters will need to migrate to KRaft first within the Strimzi 0.4x and 0.4y versions and Kafka 3.9.
+
+4) The Strimzi releases after 0.4z will continue with the 1-2 months release cycles, as usual, and will continue to adopt the Kafka patch releases.
+   Once Apache Kafka 4.1 is released, Strimzi will adopt Kafka 4.1 and continue its releases with support for Kafka 4.0 and 4.1.
+
+### Enforcing the migration to KRaft
+
+There is no way we can enforce the migration to KRaft or do it automatically.
+Users have to create the controller nodes and initiate the migration.
+This cannot be done automatically, as we would not know what the right configuration for the KRaft controller should be (storage, resources, affinity, tolerations, ...).
+
+To try to get users to migrate to KRaft in time, we will produce warning log messages and warning conditions in the `.status` section in the last Strimzi version supporting ZooKeeper (0.4x and 0.4y).
+
+In Strimzi 0.4z and newer versions we will check the Kafka clusters for the metadata type they use.
+For any cluster not using KRaft it will immediately raise an error instead of trying to reconcile the cluster.
+The check will be done based on the `Kafka` CR `.status` section.
+Users will have to downgrade the operator, migrate the cluster and upgrade the operator again.
+
+Newly deployed Kafka clusters will be expected to be KRaft only.
+
+### Extended support for Kafka 3.9
```

Comment:
can you add in this paragraph that the Strimzi version with this support will be the 0.4y release that you mentioned before? Just to make it clearer.